feat: custom precision erc20 contract

bekauz opened the pull request:
closes #326 

# Description

This pr generalizes the MockERC20 contract to accept a precision (decimals) parameter for the e2e testing suite.

In addition to that, the `eth_vault` flow is now updated to use this erc20 with custom precision (6) as the vault deposit token.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Vault and strategy modules now support configurable ERC20 token decimal precision, enabling accurate handling of tokens with different decimal places.

- **Bug Fixes**
  - Vault token precision updated to match deposit token precision, ensuring correct accounting for 6-decimal tokens.
  - Balance checks and thresholds in routing logic adjusted for improved reliability and observability.

- **Configuration**
  - Updated contract addresses, account addresses, denominations, and gRPC ports in configuration files for improved integration and compatibility.

- **Logging & Observability**
  - Enhanced informational logging in strategy and routing modules for better monitoring of balances and operations.

- **Tests**
  - MockERC20 tokens in Solidity and Rust tests now support custom decimal precision, improving test accuracy for various token types.

- **Refactor**
  - Internal method signatures updated to support token decimal configuration and improved redemption rate calculations.
  - Several methods made private to better encapsulate functionality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

keyleu commented on solidity/test/mocks/MockERC20Usdc.sol:
```diff
@@ -0,0 +1,21 @@
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+
+contract MockERC20 is ERC20, ERC20Burnable {
```
nit: Let's call it MockUSDC? or maybe we can create a MockERC20WithCustomDecimals

and provide this in the constructor and the function takes it from the state?

bekauz replied:
yeah I like the idea of `MockERC20WithCustomDecimals`. Seems like the denom precisions on eth usually track the origin asset (e.g. wbtc is precision 8), so these custom precision mocks will come in handy.

keyleu commented on solidity/test/mocks/MockERC20Usdc.sol:
Can we keep the same name here as the contract?

bekauz replied:
will address with the custom decimals erc20

bekauz commented:
@keyleu I removed the mock erc20 usdc contract and updated the existing `MockERC20` to accept configurable decimals. Updated a bunch of solidity tests so pls give that a look!
